core/src/main/java/org/apache/iceberg/io/ContentCache.java

```diff
@@ -147,10 +147,23 @@ public InputFile tryCache(InputFile input) {
     return input;
   }
 
+  /**
+   * Invalidate the cache entry for the given key.
+   *
+   * <p>Note: if there is ongoing load, this is a blocking operation, i.e. it will wait for the load
+   * to complete before invalidating the entry.
+   */
   public void invalidate(String key) {
     cache.invalidate(key);
   }
 
+  /**
+   * @deprecated will be removed in 1.7; This method does only best-effort invalidation and is
```

Comment:
Do we have to remove in 1.7 or 2.0 ?

Reply:
good question. I noticed some APIs deprecated since 1.6.0 and to be removed in 1.7.0, but I don't know the rules governing this. @Fokko do you want to advise here?

Reply:
https://iceberg.apache.org/contribute/?h=contribute#semantic-versioning

Already documented above. For core module, minor version deprecation is enough.

nit: I would use 1.7.0 instead of 1.7

Reply:
> nit: I would use 1.7.0 instead of 1.7

the rest of the file uses `1.7`,  i wanted to be consistent.
i can change as a follow-up

Reply:
I think 1.7.0 is fine. When we decide to go to a 2.x release, we can still clean up the 1.7.0 deprecations.

Reply:
OK, fair enough. I was just wondering.

Reply:
changed "1.7" to say "1.7.0"